algebird-core/src/main/scala/com/twitter/algebird/HyperLogLog.scala

```diff
@@ -294,6 +353,19 @@ case class SparseHLL(bits: Int, maxRhow: Map[Int, Max[Byte]]) extends HLL {
         buffer.update(idx, buffer(idx) max (maxb.get))
     }
   }
+
+  override def downsize(reducedBits: Int): HLL = {
+    require(reducedBits > 3 && reducedBits < bits, s"Use at least 4, but less than $bits bits")
+    val reducedSize = 1 << reducedBits
+    val reducedMaxRhoW = collection.mutable.Map.empty[Int, Max[Byte]]
+    for (i <- 0 until reducedSize) {
```

Comment:
this looks expensive for the sparse case. Can't we instead iterate over the keys that are present instead of all the buckets?